collector/CMakeLists.txt

```diff
@@ -31,6 +31,12 @@ if(NOT BPF_DEBUG_MODE)
 	set(BPF_DEBUG_MODE OFF)
 endif()
 
+if (NOT COLLECTOR_VERSION)
+	set(COLLECTOR_VERSION "0.0.0")
+endif()
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Version.h.in ${CMAKE_CURRENT_BINARY_DIR}/Version.h)
```

Comment:
From the [cmake docs](https://cmake.org/cmake/help/latest/command/configure_file.html):
```
If the input file is modified the build system will re-run CMake to re-configure the file and generate the build system again. The generated file is modified and its timestamp updated on subsequent cmake runs only if its content is changed.
```

So it will retrigger if you change the file or manually re-run cmake. The way we use it right now running with our make files, this file gets re-created on every `make image` call.